lib/compat.php

```diff
@@ -460,3 +460,20 @@ function gutenberg_render_block_with_assigned_block_context( $pre_render, $parse
  * @see WP_Block::render
  */
 remove_action( 'enqueue_block_assets', 'wp_enqueue_registered_block_scripts_and_styles' );
+
+
+/**
+ * Extends block editor settings to include a list of values for padding.
+ *
+ * @param array $settings Default editor settings.
+ *
+ * @return array Filtered editor settings.
+ */
+function gutenberg_add_line_height_presets( $settings ) {
+	list( $editor_line_height ) = (array) get_theme_support( 'editor-line-height' );
```

Comment:
do you think we should add new theme support or just rely on theme.json?

My opinion for the moment is that we should stop adding new theme supports and focus on making theme.json good enough.

Reply:
I like the sound of what you say, however, I worry about what we'll ship to WordPress 5.5 -- once there, it's difficult to change. For context, this is the full picture of style attributes:

Style prop | Status | Opt-in / Opt-out | How it works | What presets it uses
--- | --- | ---- | --- | --- 
Existing styles (font-size, color, gradient). | Released to core, stable. | Opt-out via `disable-custom-{preset-name}` | Presets: classes / Custom: inline styles with raw values | Each style prop defines its own preset.
line-height | Plugin-only, stable. | Opt-out via `disable-custom-line-height` | Presets & Custom: inline styles with raw values | None yet (this PR adds line-height presets).
padding | Plugin-only, experimental. | Opt-in via `experimental-custom-spacing` | Presets & Custom: inline styles with raw values. | None yet. WIP at https://github.com/WordPress/gutenberg/pull/23176
units (for padding) | Plugin-only, experimental. | Opt-in via `experimental-custom-units` | N/A | N/A
link color | Plugin-only, experimental. | Opt-in via `experimental-link-color` | Presets: inline styles with css variables / Custom: inline styles with raw values | Uses color's preset.

I'd be open to not adding presets for padding via the conventional mechanism (add_theme_support) because it's experimental, but, unfortunately, I think we need to add them for line-height so themes can add their own.

Reply:
Also for context! I only have a bit more than a week of work before 5.5 beta and my goals are: 1) adding presets for line-height & padding, 2) consolidating the way style attributes are stored in the markup, 3) prepare core patches.

For 2, I was wanting to engage you and @ItsJonQ  for a convo after this PR. The link color mechanism (always inline styles: CSS vars for presets, raw values for custom) works and gives us a nice method to tweak the values. However, I'm concerned about how that would work for the color status (visited, active, hover). I don't see how we can implement them using inline styles (I had [this convo with Jorge as well](https://github.com/WordPress/gutenberg/pull/22722)).

Reply:
https://github.com/WordPress/gutenberg/pull/23177/commits/efe4c26bcc69fa8e0e64f48b2ae28bfb1eb23888 exposes the line-height preset as CSS variables like the rest of preset categories do.